espresso/environment/3_2_espresso_deterministic_state_test.go

```diff
@@ -105,24 +128,231 @@ func TestDeterministicDerivationExecutionState(t *testing.T) {
 			t.Fatalf("Waiting for L2 tx:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
 		}
 
-		// Get latest safe blocks from caff node first
-		// as caff node usually lags behind the sequencer node on safe blocks due to submitting additionally to Espresso.
-		// We use l2BlockRefByLabel to get the states as the engine state will be reflected in the block.
-		caffBlock, err := caffNodeL2Client.L2BlockRefByLabel(ctx, eth.Safe)
+		// When it is the attack round, send some Espresso transactions with fakeBatcherPrivateKey directly to Espresso.
+		// Use the same way as creating a real transaction but a fake batcher private key to create a fake Espresso transaction.
```

Comment:
The L2 batch embedded in the Espresso is well formed but will be ignored as the corresponding Espresso transaction is not signed by the batcher and the batch information is not authenticated to the batch authentication contract.